samples/client/echo_api/typescript-axios/build/base.ts

```diff
@@ -53,7 +53,7 @@ export class BaseAPI {
     constructor(configuration?: Configuration, protected basePath: string = BASE_PATH, protected axios: AxiosInstance = globalAxios) {
         if (configuration) {
             this.configuration = configuration;
-            this.basePath = configuration.basePath || this.basePath;
+            this.basePath = configuration.basePath ?? this.basePath;
```

Comment:
should it be the following instead?
```
this.basePath = basePath ?? configuration.basePath
```

if basePath is provided via the constructor, then use it
otherwise use configuration.basePath instead.

Reply:
The constructor's `basePath` parameter has a default value and will never be null under any circumstances.

Reply:
Ah ok. thanks for pointing it out. I'm no expert in TS :)

so if someone pass a value for `basePath` via the constructor, it will only be used when configuration.basePath is null?

Reply:
Indeed it is.

Reply:
The question in my mind is if the basePath is already (or should be) provided in the configuration object, why we need `protected basePath: string = BASE_PATH` in the constructor?

shouldn't the constructor simply uses BASE_PATH if configuration.basePath is null?

again I'm no expert in TS but still I don't understand the logic. would appreciate if you or someone else can explain a bit more.

Reply:
Since `configuration` is an optional parameter, the `if` statement will not be entered when `configuration` is `undefined`. In this case, `this.basePath` will automatically be set to `BASE_PATH`.

Reply:
ok, thanks for the explanation